model/riscv_insts_zbc.sail

```diff
@@ -60,11 +52,7 @@ mapping clause assembly = RISCV_CLMULR(rs2, rs1, rd)
   <-> "clmulr" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
 
 function clause execute (RISCV_CLMULR(rs2, rs1, rd)) = {
-  let rs1_val = X(rs1);
-  let rs2_val = X(rs2);
-  var result : xlenbits = zeros();
-  foreach (i from 0 to (xlen_val - 1))
-    if rs2_val[i] == bitone then result = result ^ (rs1_val >> (xlen_val - i - 1));
-  X(rd) = result;
+  let prod = carryless_mul(reverse(X(rs1)), reverse(X(rs2)));
```

Comment:
I wonder if it would make sense to have a clmulr function to match the pseudocode in the spec? Otherwise the note in the spec about it being equivalent is going to read kind of weirdly.

The would be a really good application for an SMT property to verify they really are equivalent.

Reply:
I'm leaning towards leaving this as it is for now, since the code is quite clear.  We could deal with the feedback from spec integration when we actually get the Sail into the spec.

Reply:
I guess it makes me a little nervous since we don't have any tests for it and it's definitely not an obvious transformation...

Reply:
The upstream `riscv-software-src/riscv-tests` do have tests for these `cmul` variants, as I said in my PR.  I tweaked  `carryless_mul` and the tests did fail.  A property test would indeed be nice, but perhaps in a separate PR?

Reply:
Ah ok fair enough. Yeah I'll open an issue.

Reply:
#824